### PR TITLE
 Add mutationobserver for more menu

### PR DESCRIPTION
**Pull request by LunarTwilight**
This JS will hide the menu to begin with, and then if something is added it will show the menu, and if something is removed (and the menu is empty), it will hide the menu.

### CODE REVIEW

**Comment by MusikAnimal** [comment]
Sorry it's taken so long to get to this!

Could we check to see if MutationObserver is supported by the browser?

File: MoreMenu.js [hunk 1]
```diff
@@ -784,6 +784,25 @@
 				addListeners();
 			}
 			
+			if ($('.client-js .skin-vector #p-cactions').length) {
```
Comment (MusikAnimal) [inline MoreMenu.js]: maybe add a `&& typeof MutationObserver === "function"`

Comment (LunarTwilight) [reply MoreMenu.js]: Done in cbfc3a1

**Comment by LunarTwilight** [comment]
@MusikAnimal ok,updated

**Comment by MusikAnimal** [comment]
@LunarTwilight I want to make it clear I haven't forgotten about this! I have the tab pinned in my browser. I have been swamped with other priorities, and I'll need to do a lot of testing for this change. I hope to get to it sometime next week. Apologies for the long wait, and for encouraging you to create a PR only to leave it open for almost four months... :/

**Comment by LunarTwilight** [comment]
It's fine~ (I mostly pinged you because I wasn't sure if you were aware I updated my edit based on your review, not because I wanted to poke you about it :))

**Comment by MusikAnimal** [comment]
@LunarTwilight I finally got around to testing this and seems to work! I have one concern... this can cause "jumping" of the top nav. So if the More menu is removed, the "View history" tab, etc., shifts over quite a bit. This will cause misfires where someone clicks on the wrong thing if they're navigating quickly. We made a peer gadget to prevent this sort of jumpiness (see [meta:MoreMenu#Setup](https://meta.wikimedia.org/wiki/MoreMenu#Setup)). I'm not sure what to do... maybe it's best to just leave the items in the More menu? :-/

**Comment by LunarTwilight** [comment]
Hm, when does it jump for you? It didn't jump for me, but I pasted it into my console.

**Comment by MusikAnimal** [comment]
You need to first ensure every item in the More menu gets moved to User/Page (I think you knew that much). In my case there was only a "Move" link, and that got moved under Page. So the scenario is that initial More menu exists on page load (because it's put there by MediaWiki), then the script initializes and removes the More menu. That's when it jumps.

To test it, add MoreMenu as a user script. Be sure to also set up the CSS that prevents the jumping of the User/Page menus themselves: https://meta.wikimedia.org/wiki/MoreMenu#User_installation

**Comment by LunarTwilight** [comment]
Hm, for me it seems to jump with the CSS, but without the CSS it doesn't jump, there's just the initial jump when the more menu gets replaced.

**Comment by MusikAnimal** [comment]
The first release of the new version has been made, which uses your MutationObserver implementation, with some modifications. I am very sorry I never got your code merged... should have done that while it was still mergable! I did give credit here, though: https://github.com/MusikAnimal/MoreMenu/blob/086a39958cd24cf342a3392c322cf4957ebd46b6/src/MoreMenu.js#L788

Thanks again for your work, and extreme patience! I am going to close this PR as it is for the old version. Kind regards

**Comment by LunarTwilight** [comment]
No problem, just glad to see it made it in :+1:

**Pull request closed without merging**